build_runner/test/package_graph/package_graph_test.dart

```diff
@@ -150,6 +150,63 @@ void main() {
           throwsA(anything));
     });
   });
+
+  group('orderedPackages', () {
```

Comment:
Please add another test for the shared dependencies scenario I listed above.

Reply:
Done